coding-the-legislation/40_legislation_evolutions.md

```diff
@@ -60,67 +61,71 @@ we get the output:
 
 [Read more about how to code parameters](./legislation_parameters.md#parameters-and-time).
 
-## Variable defined only for a specific time intervale
+## Variable defined until a specific date
 
-As the legislation evolves, some fiscal or benefit mechanism appear and disapear. When creating a variable, you can specify the attribute `start_date` and `stop_date` to defines in which time intervale this variable make sense.
+As the legislation evolves, some fiscal or benefit mechanism appear and disapear.  
+For every variable, you can specify the attribute `end`; it defines until when this variable makes sense.
 
-When called outside of its definition intervale, a variable will **not** execute its formula and instead **return its default value**.
+If called outside of its definition time, a variable will **not** execute its formula and instead **return its default value**.
 
 For instance:
 ```py
 class flat_tax_on_salary(Variable):
     column = FloatCol
     entity = Person
     label = u"Individualized and monthly paid tax on salaries"
-    start_date = date(2014, 01, 01)
+    end = '2014-12-31'
     definition_period = MONTH
 
-    def function(person, period, legislation):
-        ...
-
-class progressive_income_tax(Variable):
-    column = FloatCol
-    entity = Person
-    label = u"Former tax replaced by the flat tax on the 1st of Jan 2014"
```

Comment:
Yep, this flat tax was supposed to change in 2017, it can be repealed in 2014 🙂 !